src/Agent.Worker/Build/TfsVCSourceProvider.cs

```diff
@@ -405,7 +408,15 @@ public sealed class TfsVCSourceProvider : SourceProvider, ISourceProvider
                         // Cleanup the comment file.
                         if (File.Exists(commentFile))
                         {
-                            File.Delete(commentFile);
+                            try
+                            {
+                               await IOUtil.DeleteFileWithRetry(commentFile, cancellationToken);
+                            }
+                            catch (Exception ex)
+                            {
+                                Trace.Warning($"Unable to delete comment file, ex:{ex.GetType()}");
```

Comment:
Will it be useful/meaningful to see exception type in the outpur?

Reply:
Since it's generic exception catcher I believe it's useful to trace what was the exception if caller doesn't write that